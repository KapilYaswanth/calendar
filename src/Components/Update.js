import React,{useState} from 'react'
export default function Update(props) {
    console.log('props',props);
    const [updatedname, setupdatedname] = useState("")
    const [updatedstart, setupdatedstart] = useState("")
    const [updatedend, setupdatedend] = useState("")
    const handleClick = () => {
    var gapi = window.gapi
    var CLIENT_ID = "639100172651-75tnqt7k8b64d75n8en3ghosh8jnh9d9.apps.googleusercontent.com"
    var API_KEY = "AIzaSyBChZ90Wv3zs6WMlkgW75EZbobmbsO8ePs"
    var SCOPES = "https://www.googleapis.com/auth/calendar.events"
        gapi.load('client:auth2', () => {
          console.log('loaded client')
    
          gapi.client.init({
            apiKey: API_KEY,
            clientId: CLIENT_ID,
            scope: SCOPES,
          })
    
    
          gapi.auth2.getAuthInstance().signIn()
          .then(() => {
            var event = {
              'summary': updatedname,
              'start': {
                'date': updatedstart,
                'timeZone': 'Asia/Kolkata'
              },
              'end': {
                'date': updatedend,
                'timeZone': 'Asia/Kolkata'
              },
              'hangoutLink': 'https://meet.google.com/rph-dxck-cxu',
           }
            var request = gapi.client.calendar.events.update({
              'calendarId': 'primary',
              'eventId':props.id,
              'resource': event,
            })
    
            request.execute(event => {
              console.log(event)
            })
          })
        })
      }
    return (
        
        <div>
            <form data-testid="update form" onSubmit={(event)=>{
                event.preventDefault();
                handleClick();
                setupdatedname("");
                setupdatedstart("");
                setupdatedend("");
            }} >
        <input type="text" data-testid="Update event name" value={updatedname} id="outlined-basic" placeholder="Event name" onChange={(e)=>setupdatedname(e.target.value)} required/><br/>
        <input type="text" data-testid="Update start date" value={updatedstart} id="outlined-basic" placeholder="Start date" variant="outlined" onChange={(e)=>setupdatedstart(e.target.value)}required/><br/>
        <input type="text" data-testid="Update end date" value={updatedend} id="outlined-basic" placeholder="End date" variant="outlined" onChange={(e)=>setupdatedend(e.target.value)}required/>
        <div><button type="submit">Update Event</button></div>
      </form>
      
        </div>
    )
}
