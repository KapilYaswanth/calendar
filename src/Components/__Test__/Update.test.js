import React from 'react';
import {render,fireEvent, cleanup} from '@testing-library/react'; 
import renderer from 'react-test-renderer'
import Update from '../Update';

afterEach(cleanup);

it("renders correctly",()=>{
    const {queryByTestId} = render(<Update />)
    expect(queryByTestId("update form")).toBeTruthy();
})

describe("Updated Event name",() => {
    it("updates on change",() => {
        const {queryByTestId} = render(<Update />)
        const textField = queryByTestId("Update event name");
        fireEvent.change(textField,{target:{value:"test"}})
        expect(textField.value).toBe("test")
    })
})

describe("Updated Start Date",() => {
    it("updates on change",() => {
        const {queryByTestId} = render(<Update />)
        const textField = queryByTestId("Update start date");
        fireEvent.change(textField,{target:{value:"test"}})
        expect(textField.value).toBe("test")
    })
})

it("Update Form Matches Snapshot",()=>{
    const tree = renderer.create(<Update/>).toJSON();
    expect(tree).toMatchSnapshot();
}) 