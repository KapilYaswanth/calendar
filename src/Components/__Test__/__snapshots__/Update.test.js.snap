// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Form Matches Snapshot 1`] = `
<div>
  <form
    data-testid="update form"
    onSubmit={[Function]}
  >
    <input
      data-testid="Update event name"
      id="outlined-basic"
      onChange={[Function]}
      placeholder="Event name"
      required={true}
      type="text"
      value=""
    />
    <br />
    <input
      data-testid="Update start date"
      id="outlined-basic"
      onChange={[Function]}
      placeholder="Start date"
      required={true}
      type="text"
      value=""
      variant="outlined"
    />
    <br />
    <input
      data-testid="Update end date"
      id="outlined-basic"
      onChange={[Function]}
      placeholder="End date"
      required={true}
      type="text"
      value=""
      variant="outlined"
    />
    <div>
      <button
        type="submit"
      >
        Update Event
      </button>
    </div>
  </form>
</div>
`;
